tests/aws/services/secretsmanager/test_secretsmanager.py

```diff
@@ -279,51 +280,55 @@ def test_list_secrets_filtering(self, aws_client, create_secret):
         self._wait_created_is_listed(aws_client.secretsmanager, secret_id=secret_name_3)
         self._wait_created_is_listed(aws_client.secretsmanager, secret_id=secret_name_4)
 
-        def assert_secret_names(res, include_secrets, exclude_secrets):
-            for secret in res["SecretList"]:
-                assert secret["Name"] in include_secrets
-                assert secret["Name"] not in exclude_secrets
+        def assert_secret_names(res: dict, include_secrets: set[str], exclude_secrets: set[str]):
+            secret_names = {secret["Name"] for secret in res["SecretList"]}
+            assert (
+                include_secrets - secret_names
+            ) == set(), "At least one secret which should be included is not."
+            assert (
+                exclude_secrets - secret_names
+            ) == exclude_secrets, "At least one secret which should not be included is."
```

Comment:
nice fix using set semantics here. But is there a reason why we don't just do a snapshot test here?

Reply:
The problem is that we really want to make this test resilient against any other secrets being left over from other tests (because that's what broke the first two tries). The filters are very generic and a leftover secret could leak into a snapshot / response. This is why this assert really just checks if certain secrets are in the response and others are not (while leaving the possibility of other secrets being in the response).